File: isa2/demo/src/main/java/isa2/demo/DTO/AdventureDTO.java
package isa2.demo.DTO;

import lombok.Getter;
import lombok.Setter;

import java.util.Collection;
import java.util.Set;

@Getter
@Setter
public class AdventureDTO {
    private Integer id;
    private Integer adventureOwnerId;
    private String name;
    private AddressDTO addressDTO;
    private String description;
    private String instructorBio;
    private Double avgGrade;
    private Set<String> photos;
    private Integer maxNumberOfGuests;
    private String houseRules;
    private String defaultFishingEquipment;
    private Double pricePerDay;
    private Integer cancellationFee;
    private String entityPhoto;
    private Collection<ReservationDTO> reservations;
    private Collection<AdditionalServiceDTO> additionalServices;
}

File: client/src/app/model/adventure-dto.ts
import {Address} from './address';
import {AdditionalService} from './additional-service';
import {AddressDTO} from './address-dto.model';
import {RoomDTO} from "./room-dto.model";
import {ReservationDTO} from "./reservation-dto.model";
import {AdditionalServicesDTO} from "./AdditionalServicesModel/additional-services-dto.model";

export class AdventureDTO{
  public id!: number;
  public adventureOwnerId!: number;
  public name!: string;
  public addressDTO: AddressDTO;
  public description!: string;
  public instructorBio!: string;
  public photos: string[] = [];
  public maxNumberOfGuests!: number;
  public houseRules!: string;
  public additionalServices!: AdditionalServicesDTO[];
  public pricePerDay!: number;
  public cancellationFee!: number;
  public entityPhoto!: string;
  public defaultFishingEquipment!: string;
  public reservations: ReservationDTO[];
  public averageGrade!: number;

  constructor(id: number, name: string, description: string, photos: string[], adventureOwnerId: number,
              maxNumberOfGuests: number, pricePerDay: number,
              addressDTO: AddressDTO, reservationsDTO: ReservationDTO[], additionalServices: AdditionalServicesDTO[]) {
    this.id = id;
    this.adventureOwnerId = adventureOwnerId;
    this.name = name;
    this.description = description;
    this.photos = photos;
    this.maxNumberOfGuests = maxNumberOfGuests;
    this.pricePerDay = pricePerDay;
    this.addressDTO = addressDTO;
    this.reservations = reservationsDTO;
    this.additionalServices = additionalServices;
  }
}
